### PR TITLE
add cars_list db,  show: table, pagination

### PR DESCRIPTION
**Pull request by ViktorBurchenia**

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -1,2 +1,3 @@
 // Custom bootstrap variables must be set or imported *before* bootstrap.
-@import "bootstrap"; 
\ No newline at end of file
+@import "bootstrap"; 
+@import 'search'
```
Comment (AsakuraRET) [inline app/assets/stylesheets/application.scss]: Add new line please

File: app/assets/stylesheets/search.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.search-form {
+  display: flex;
+  flex-direction: column;
+  max-width: 400px;
+  grid-gap: 10px;
+}
+
```
Comment (AsakuraRET) [inline app/assets/stylesheets/search.scss]: Add new line please for end file

File: app/controllers/cars_controller.rb [hunk 3]
```diff
@@ -1,9 +1,19 @@
 class CarsController < ApplicationController
   before_action :set_car, only: %i[ show edit update destroy ]
 
+SORTING = {
+  'date_added_asc' => {date_added: :asc},
+  'date_added_desc' => {date_added: :desc},
+  'price_asc' => {price: :asc},
+  'price_desc' => {price: :desc}
+}
```
Comment (AsakuraRET) [inline app/controllers/cars_controller.rb]: Please add .freeze for constant

File: app/controllers/cars_controller.rb [hunk 4]
```diff
@@ -1,9 +1,19 @@
...
+
+
   # GET /cars or /cars.json
   def index
     @cars = Car.all.page params[:page]
+    @cars_total = Car.all
+    @cars = @cars.order(**SORTING[params[:sort_by]]) if params[:sort_by].present?
```
Comment (AsakuraRET) [inline app/controllers/cars_controller.rb]: Please use private def for this sorting and then use
def_name if params[:sort_by].present?

File: app/views/cars/index.html.erb [hunk 5]
```diff
@@ -1,14 +1,78 @@
...
+          </tr>
+         <% end %>
+        </tbody>
+      </table>
 </div>
 
-<%= link_to "New car", new_car_path %>
+<%= link_to new_car_path do %>
+   <%= form_with  do |f|%>
+    <%=f.submit 'New car'%>
```
Comment (AsakuraRET) [inline app/views/cars/index.html.erb]: Please add whitespace

File: app/views/cars/index.html.erb [hunk 6]
```diff
@@ -1,14 +1,78 @@
...
+    <% end %>
+<% end %>
+
+<div class ='row'>
+<div class ='col-7 offset-5'>
+<%= paginate @cars %>
+<div>
+<div>
```
Comment (AsakuraRET) [inline app/views/cars/index.html.erb]: Add newline

File: app/views/cars/index.html.erb [hunk 7]
```diff
@@ -1,14 +1,78 @@
...
-<%= link_to "New car", new_car_path %>
+<%= link_to new_car_path do %>
+   <%= form_with  do |f|%>
+    <%=f.submit 'New car'%>
+    <% end %>
+<% end %>
+
+<div class ='row'>
+<div class ='col-7 offset-5'>
+<%= paginate @cars %>
```
Comment (AsakuraRET) [inline app/views/cars/index.html.erb]: Please add whitespace

File: app/views/cars/new.html.erb [hunk 8]
```diff
@@ -5,5 +5,11 @@
 <br>
 
 <div>
-  <%= link_to "Back to cars", cars_path %>
+  <%= link_to cars_path do %>
+   <%= form_with  do |f|%>
+    <%=f.submit 'Back to cars'%>
+    <% end %>
```
Comment (AsakuraRET) [inline app/views/cars/new.html.erb:12]: Please remove whitespace

File: app/views/search/search_car.html.erb [hunk 9]
```diff
@@ -0,0 +1,66 @@
+  <p style="color: green"><%= notice %></p>
+  <h1>Search</h1>
+
+<%= form_with url: search_path, method: :get, data: {turbo: false}, class: 'search-form' do |form|%>
+
+
```
Comment (AsakuraRET) [inline app/views/search/search_car.html.erb]: Remove unused lines

File: app/views/search/search_car.html.erb [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+<% end %>
+
+<%-if @cars != nil%>
+    <div id="cars">
+          <table class="table table-bordered table-hover table-fixed">
+            <thead>
+              <tr>
+                <th scope="col">id</th>
+                <th scope="col">make</th>
+                <th scope="col">model</th>
```
Comment (AsakuraRET) [inline app/views/search/search_car.html.erb]: Please align nesting, use 2 spaces or 1 tab

File: app/views/search/search_car.html.erb [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+                    <td>
+                      <%= link_to "Show this car", car %>
+                    </td>
+
+              </tr>
+            <% end %>
+            </tbody>
+          </table>
+    </div>
+<% end %>
```
Comment (AsakuraRET) [inline app/views/search/search_car.html.erb]: Please add newline to end file

**Review by AsakuraRET — CHANGES_REQUESTED** [review]

File: app/controllers/cars_controller.rb (hunk 3 above)
Comment (ViktorBurchenia) [reply app/controllers/cars_controller.rb]: fixed

File: app/views/cars/index.html.erb (hunk 6 above)
Comment (ViktorBurchenia) [reply app/views/cars/index.html.erb]: fixed

File: app/views/cars/index.html.erb (hunk 7 above)
Comment (ViktorBurchenia) [reply app/views/cars/index.html.erb]: fixed

File: app/views/cars/index.html.erb (hunk 5 above)
Comment (ViktorBurchenia) [reply app/views/cars/index.html.erb]: Fixed

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (ViktorBurchenia) [reply app/assets/stylesheets/application.scss]: Fixed

File: app/assets/stylesheets/search.scss (hunk 2 above)
Comment (ViktorBurchenia) [reply app/assets/stylesheets/search.scss]: Fixed

File: app/views/cars/new.html.erb (hunk 8 above)
Comment (ViktorBurchenia) [reply app/views/cars/new.html.erb:12]: Fixed

File: app/views/search/search_car.html.erb (hunk 11 above)
Comment (ViktorBurchenia) [reply app/views/search/search_car.html.erb]: Fixed

File: app/views/search/search_car.html.erb (hunk 10 above)
Comment (ViktorBurchenia) [reply app/views/search/search_car.html.erb]: Fixed

File: app/views/search/search_car.html.erb (hunk 9 above)
Comment (ViktorBurchenia) [reply app/views/search/search_car.html.erb]: Fixed

**Review by AsakuraRET — APPROVED** [review]

**Pull request merged**